src/WebJobs.Script/Rpc/RpcInitializationService.cs

```diff
@@ -150,9 +150,19 @@ private Task InitializeWebHostRuntimeChannelsAsync()
             return Task.CompletedTask;
         }
 
-        private bool ShouldStartInPlaceholderMode()
+        internal bool ShouldStartInPlaceholderMode()
         {
-            return string.IsNullOrEmpty(_workerRuntime) && _environment.IsPlaceholderModeEnabled();
+            if (string.IsNullOrEmpty(_workerRuntime) && _environment.IsPlaceholderModeEnabled())
+            {
+                if (_environment.IsLinuxHostingEnvironment())
```

Comment:
Rather than special casing linux like this, can you do: if (_environment.IsAppServiceEnvironment() && _environment.IsPlaceholderTemplateSite()) { return true; }? That way we're

Reply:
This change is only for windows as template sites are only valid for windows placeholders.